### PR TITLE
[Feature] Support FileIO cache for iceberg table

### LINKED ISSUE #6235: add iceberg fileIO cache for iceberg external table
## Feature request

**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

under my test case,   select from iceberg external table,  fe call iceberg's planFile interface cost much time.
so I think need to add a cache for iceberg to get metadata from remote storage.

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by hongli-my**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes  #6235

## Problem Summary(Required) ：

I use   this pr: https://github.com/apache/iceberg/pull/4518
and copy  `ByteBufferInputStream`,    `MultiBufferInputStream` , `SingleBufferInputStream` class, becase iceberg master branch  hava implements this interface `SeekableInputStream` .

I only expose `fileIO.cache.max-total-bytes`  to limit max cache size and to evict data if reach this limit.

### CODE REVIEW

**Comment by DorianZheng** [comment]
@hongli-my Plz sign the CAL

File: fe/fe-core/src/main/java/com/starrocks/external/iceberg/io/IcebergCachingFileIO.java
Comment (hongli-my) [reply fe/fe-core/src/main/java/com/starrocks/external/iceberg/io/IcebergCachingFileIO.java:30]: done

**Comment by hongli-my** [comment]
for  starrocks_fe_unittest : testWithResourceName run  on my env:      is no this table(because my env no this table), but the  github fe unittest  error is  can not connect to metastore.  
```
2022-05-20 14:56:10 [main] INFO  HiveMetaStoreThriftClient:416 - Trying to connect to metastore with URI thrift://VM-72-127-centos:9083
2022-05-20 14:56:10 [main] INFO  HiveMetaStoreThriftClient:496 - Opened a connection to metastore, current connections: 1
2022-05-20 14:56:10 [main] INFO  HiveMetaStoreThriftClient:551 - Connected to metastore.
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 2.991 s <<< FAILURE! - in com.starrocks.catalog.IcebergTableTest
[ERROR] testWithResourceName(com.starrocks.catalog.IcebergTableTest)  Time elapsed: 2.874 s  <<< ERROR!
com.starrocks.external.iceberg.StarRocksIcebergException: Failed to load Iceberg table with id: db0.table0
	at com.starrocks.catalog.IcebergTableTest.testWithResourceName(IcebergTableTest.java:94)
Caused by: org.apache.iceberg.exceptions.NoSuchTableException: Table does not exist: db0.table0
	at com.starrocks.catalog.IcebergTableTest.testWithResourceName(IcebergTableTest.java:94)
```

File: fe/fe-core/src/main/java/com/starrocks/external/iceberg/io/IcebergCachingFileIO.java
Comment (caneGuy) [inline fe/fe-core/src/main/java/com/starrocks/external/iceberg/io/IcebergCachingFileIO.java:89]: we may get outdated file if we never touch the maxTotalBytes, am i right?

Comment (hongli-my) [reply fe/fe-core/src/main/java/com/starrocks/external/iceberg/io/IcebergCachingFileIO.java:89]: this cache the iceberg meta file,    if  delete  record,  add record, or update in iceberg, will  add new meta file in iceberg

File: fe/fe-core/src/main/java/com/starrocks/external/iceberg/io/IcebergCachingFileIO.java
Comment (caneGuy) [inline fe/fe-core/src/main/java/com/starrocks/external/iceberg/io/IcebergCachingFileIO.java:158]: nit: newCacheEntry or createCacheEntry

**Review by caneGuy — COMMENTED** [review]
other LGTM, pending on UT regression

File: fe/fe-core/src/main/java/com/starrocks/external/iceberg/io/IcebergCachingFileIO.java
Comment (hongli-my) [reply fe/fe-core/src/main/java/com/starrocks/external/iceberg/io/IcebergCachingFileIO.java]: done

**Comment by caneGuy** [comment]
run starrocks_fe_unittest

**Review by caneGuy — APPROVED** [review]

**Review by caneGuy — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/external/iceberg/IcebergUtil.java
Comment (caneGuy) [inline fe/fe-core/src/main/java/com/starrocks/external/iceberg/IcebergUtil.java:60]: This api has been changed, so the unit test will exit with failure it break the mock semantics below in IcebergTableTest
```
new MockUp<IcebergUtil>() {
            @Mock
            public IcebergCatalog getIcebergHiveCatalog(String uris) {
                return icebergCatalog;
            }
        };
```

**Comment by caneGuy** [comment]
Plz fix the unit test of IcebergTable @hongli-my thanks

**Comment by hongli-my** [comment]
```
Error Message
Missing 1 invocation to:
com.starrocks.server.GlobalStateMgr#getCurrentSystemInfo()
Stacktrace
Missing 1 invocation to:
com.starrocks.server.GlobalStateMgr#getCurrentSystemInfo()
Caused by: Missing invocations
	at com.starrocks.task.AgentBatchTask.run(AgentBatchTask.java:160)
	at java.lang.Thread.run(Thread.java:750)
```

i run  this command local `mvn test -Dtest=RollupJobV2Test  -DskipTests=false -DfailIfNoTests=false`, everything is ok, 

can i run starrocks_fe_unittest local?

**Comment by DorianZheng** [comment]
run starrocks_fe_unittest

**Comment by DorianZheng** [comment]
run starrocks_fe_unittest

File: fe/fe-core/src/main/java/com/starrocks/catalog/IcebergTable.java
Comment (DorianZheng) [inline fe/fe-core/src/main/java/com/starrocks/catalog/IcebergTable.java:169]: why do we need to throw exception on unknown properties?

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 80 / 341 (23.46%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/iceberg/io/MultiBufferInputStream.java|0|154|00.00%|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|0|3|00.00%|
|:large_blue_circle:|com/starrocks/external/iceberg/io/SingleBufferInputStream.java|9|70|12.86%|
|:large_blue_circle:|com/starrocks/external/iceberg/io/ByteBufferInputStream.java|3|13|23.08%|
|:large_blue_circle:|com/starrocks/catalog/IcebergTable.java|4|8|50.00%|
|:large_blue_circle:|com/starrocks/external/iceberg/IcebergUtil.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/external/iceberg/io/IOUtil.java|9|14|64.29%|
|:large_blue_circle:|com/starrocks/external/iceberg/io/IcebergCachingFileIO.java|51|74|68.92%|
|:large_blue_circle:|com/starrocks/external/iceberg/IcebergHiveCatalog.java|3|3|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/external/iceberg/io/IcebergCachingFileIO.java
Comment (DorianZheng) [inline fe/fe-core/src/main/java/com/starrocks/external/iceberg/io/IcebergCachingFileIO.java:1]: If this file is from iceberg repo too, plz add license header.

File: fe/fe-core/src/main/java/com/starrocks/external/iceberg/io/IcebergCachingFileIO.java
Comment (DorianZheng) [inline fe/fe-core/src/main/java/com/starrocks/external/iceberg/io/IcebergCachingFileIO.java:173]: when will this happen?

**Review by DorianZheng — DISMISSED** [review]
I have left some comment, others LGTM

File: fe/fe-core/src/main/java/com/starrocks/catalog/IcebergTable.java
Comment (hongli-my) [reply fe/fe-core/src/main/java/com/starrocks/catalog/IcebergTable.java]: I have change ,  I don't see that there is still operation in it  CUSTOM_CATALOG case.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 79 / 339 (23.30%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/iceberg/io/MultiBufferInputStream.java|0|154|00.00%|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|0|3|00.00%|
|:large_blue_circle:|com/starrocks/external/iceberg/io/SingleBufferInputStream.java|9|70|12.86%|
|:large_blue_circle:|com/starrocks/external/iceberg/io/ByteBufferInputStream.java|3|13|23.08%|
|:large_blue_circle:|com/starrocks/catalog/IcebergTable.java|3|6|50.00%|
|:large_blue_circle:|com/starrocks/external/iceberg/IcebergUtil.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/external/iceberg/io/IOUtil.java|9|14|64.29%|
|:large_blue_circle:|com/starrocks/external/iceberg/io/IcebergCachingFileIO.java|51|74|68.92%|
|:large_blue_circle:|com/starrocks/external/iceberg/IcebergHiveCatalog.java|3|3|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/external/iceberg/io/IcebergCachingFileIO.java
Comment (hongli-my) [reply fe/fe-core/src/main/java/com/starrocks/external/iceberg/io/IcebergCachingFileIO.java:1]: done

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 79 / 339 (23.30%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/iceberg/io/MultiBufferInputStream.java|0|154|00.00%|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|0|3|00.00%|
|:large_blue_circle:|com/starrocks/external/iceberg/io/SingleBufferInputStream.java|9|70|12.86%|
|:large_blue_circle:|com/starrocks/external/iceberg/io/ByteBufferInputStream.java|3|13|23.08%|
|:large_blue_circle:|com/starrocks/catalog/IcebergTable.java|3|6|50.00%|
|:large_blue_circle:|com/starrocks/external/iceberg/IcebergUtil.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/external/iceberg/io/IOUtil.java|9|14|64.29%|
|:large_blue_circle:|com/starrocks/external/iceberg/io/IcebergCachingFileIO.java|51|74|68.92%|
|:large_blue_circle:|com/starrocks/external/iceberg/IcebergHiveCatalog.java|3|3|100.00%|

**Review by caneGuy — APPROVED** [review]
LGTM

**Review by DorianZheng — APPROVED** [review]
LGTM

**Review by imay — APPROVED** [review]

**Pull request merged**
